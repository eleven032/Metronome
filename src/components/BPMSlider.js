import React from 'react';
import Slider from '@material-ui/core/Slider';

const defaultBPM = 80;

export default function BPMSlider() {
  return (
    <Slider
      track="normal"
      defaultValue={defaultBPM}
      aria-labelledby="discrete-slider-small-steps"
      step={5}
      min={60}
      max={240}
    />
  );
}
