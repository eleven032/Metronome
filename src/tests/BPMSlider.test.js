import React from 'react';
import { create } from 'react-test-renderer';
import BPMSlider from '../components/BPMSlider';

describe('BPMSlider Test', () => {
  test('should be a Slider tag', () => {
    const component = create(<BPMSlider />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
