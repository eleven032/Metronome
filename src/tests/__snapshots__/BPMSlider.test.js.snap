// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BPMSlider Test should be a Slider tag 1`] = `
<span
  className="MuiSlider-root MuiSlider-colorPrimary"
  onMouseDown={[Function]}
>
  <span
    className="MuiSlider-rail"
  />
  <span
    className="MuiSlider-track"
    style={
      Object {
        "left": "0%",
        "width": "11.11111111111111%",
      }
    }
  />
  <input
    type="hidden"
    value="80"
  />
  <span
    aria-labelledby="discrete-slider-small-steps"
    aria-orientation="horizontal"
    aria-valuemax={240}
    aria-valuemin={60}
    aria-valuenow={80}
    className="MuiSlider-thumb MuiSlider-thumbColorPrimary"
    data-index={0}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    role="slider"
    style={
      Object {
        "left": "11.11111111111111%",
      }
    }
    tabIndex={0}
  />
</span>
`;
